flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/functions/SystemFunctionUtils.java

```diff
@@ -108,83 +108,305 @@ public static TimestampData toTimestamp(String str, String format, String timezo
+                if (second > Integer.MAX_VALUE) {
+                    return null;
+                }
+                return (int) second / 60;
             case "HOUR":
-                return second.intValue() / 3600;
+                if (second > Integer.MAX_VALUE) {
+                    return null;
+                }
+                return (int) second / 3600;
             case "DAY":
-                return second.intValue() / (24 * 3600);
+                if (second > Integer.MAX_VALUE) {
+                    return null;
+                }
+                return (int) second / (24 * 3600);
             case "MONTH":
                 return to.get(Calendar.YEAR) * 12
-                        + to.get(Calendar.MONDAY)
-                        - (from.get(Calendar.YEAR) * 12 + from.get(Calendar.MONDAY));
+                        + to.get(Calendar.MONTH)
+                        - (from.get(Calendar.YEAR) * 12 + from.get(Calendar.MONTH));
             case "YEAR":
                 return to.get(Calendar.YEAR) - from.get(Calendar.YEAR);
             default:
-                LOG.error("Unsupported timestamp diff: {}", symbol);
-                throw new RuntimeException("Unsupported timestamp diff: " + symbol);
+                throw new RuntimeException(
+                        String.format(
+                                "Unsupported timestamp interval unit %s. Supported units are: SECOND, MINUTE, HOUR, DAY, MONTH, YEAR",
+                                timeIntervalUnit));
+        }
+    }
+
+    public static LocalZonedTimestampData timestampadd(
+            String timeIntervalUnit, Integer interval, LocalZonedTimestampData timePoint) {
+        if (interval == null || timePoint == null) {
+            return null;
+        }
+        return LocalZonedTimestampData.fromEpochMillis(
+                timestampadd(timeIntervalUnit, interval, timePoint.getEpochMillisecond()));
+    }
+
+    public static ZonedTimestampData timestampadd(
+            String timeIntervalUnit, Integer interval, ZonedTimestampData timePoint) {
+        if (interval == null || timePoint == null) {
+            return null;
+        }
+        return ZonedTimestampData.of(
+                timestampadd(timeIntervalUnit, interval, timePoint.getMillisecond()),
+                0,
+                timePoint.getZoneId());
+    }
+
+    public static TimestampData timestampadd(
+            String timeIntervalUnit, Integer interval, TimestampData timePoint) {
+        if (interval == null || timePoint == null) {
+            return null;
+        }
+        return TimestampData.fromMillis(
+                timestampadd(timeIntervalUnit, interval, timePoint.getMillisecond()));
+    }
+
+    private static long timestampadd(String timeIntervalUnit, int interval, long timePoint) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeZone(TimeZone.getTimeZone("UTC"));
+        calendar.setTime(new Date(timePoint));
```

Comment:
Can we cache a static `Calendar` and avoid creating one every time when this method got invoked?